jobrunner/run.py

```diff
@@ -188,7 +196,19 @@ def handle_job(job, api, mode=None, paused=None):
         )
         return
 
-    initial_status = api.get_status(definition)
+    try:
```

Comment:
Not one for this change, but `handle_job` is getting quite long. It took a while for me to read through and work out what was happening here. Would be worth thinking about how this function could be refactored in future.

Reply:
That's exactly what my re-thinking scheduling doc suggests :)

Basically, split the logic between two loops/functions, one that handles PENDING and one that handles RUNNING.

Reply:
But also, large functions are often unavoidable in schedulers, IME, as its essentially a state transition engine, and those often don't decompose well.